web/observing_changes/observing_changes_to_light_dom_children/my_element.dart

```diff
@@ -1,27 +1,39 @@
 @HtmlImport('my_element.html')
-library web.observing_changes.observing_changes_to_light_dom_children;
+library my_element;
 
 import 'dart:html';
-
+import 'package:web_components/web_components.dart' show HtmlImport;
 import 'package:polymer/polymer.dart';
 
-@CustomTag('my-element')
+@PolymerRegister('my-element')
 class MyElement extends PolymerElement {
-  @observable String message = '';
+  int _counter = 1;
+
+  @property String message = '';
+
+  MutationObserver observer;
+  MyElement.created() : super.created();
 
-  MyElement.created() : super.created() {
-    onMutation(this).then(childrenUpdated);
+  ready() {
+    // onMutation callback isn't supported in Polymer 1.0 anymore
```

Comment:
I believe the  ´dom-change´ event is fired only by ´dom-repeat´ and
´dom-if´ templates when they update the dom. I used it successfully in such
a use case (´dom-if´ template). I had to use `convertToDart` on the event.

Il giorno mer 21 ott 2015 alle ore 14:47 Günter Zöchbauer <
notifications@github.com> ha scritto:

> In
> web/observing_changes/observing_changes_to_light_dom_children/my_element.dart
> <https://github.com/dart-lang/polymer-dart-patterns/pull/36#discussion_r42617781>
> :
>
> >
> > -  MyElement.created() : super.created() {
> > -    onMutation(this).then(childrenUpdated);
> > +  ready() {
> > +    // onMutation callback isn't supported in Polymer 1.0 anymore
>
> I tried to implement this by listening to the dom-change event but seems
> it isn't fired by Polymer.dom(this).append() or
> Polymer.dom(this.root).append()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dart-lang/polymer-dart-patterns/pull/36/files#r42617781>
> .
>